src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -824,7 +824,25 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         assert(!node->IsEmbMaskOp());
                         if (HWIntrinsicInfo::IsExplicitMaskedOperation(intrin.id))
                         {
-                            GetEmitter()->emitIns_R_R_R(ins, emitSize, targetReg, op1Reg, op2Reg, opt);
+                            if (isRMW)
```

Comment:
can you confirm in what cases we come at this code path vs. the one at the end of this file?

Reply:
`CreateMaskForNextActiveElement` will come there, and any HW intrinsic that has two parameters with an explicit masked operation. But `CreateMaskForNextActiveElement` is the first RMW to reach this path.

Reply:
Why cannot `NI_Sve_CreateMaskForFirstActiveElement` handled here itself after you remove the `SpecialCodegen` flag for it. They both need `INS_OPTS_SCALABLE_B` as opt. So if this code can work for 
`NI_Sve_CreateMaskForNextActiveElement`, wondering why can't it work for `NI_Sve_CreateMaskForFirstActiveElement`?

Reply:
FirstActiveElement will as well, sorry I didn't include it in my comment

Reply:
Oh, I see now. Yes, we shouldn't need the SpecialCodeGen for FirstActiveElement.

Reply:
Ok, I actually tried it out. So 'pnext' is `PNEXT   <Pdn>.<T>, <Pv>, <Pdn>.<T>` while 'pfirst' is `PFIRST  <Pdn>.B, <Pg>, <Pdn>.B`. So, yea, we actually need to do SpecialCodeGen for FirstActiveElement.

Reply:
hhm, not sure I am still following. Since `FirstActiveElement` code that you have is:

```c++
                assert(isRMW);
                assert(HWIntrinsicInfo::IsExplicitMaskedOperation(intrin.id));

                if (targetReg != op2Reg)
                {
                    assert(targetReg != op1Reg);
                    GetEmitter()->emitIns_Mov(INS_sve_mov, emitTypeSize(node), targetReg, op2Reg, /* canSkip */ true);
                }

                GetEmitter()->emitIns_R_R(ins, emitSize, targetReg, op1Reg, INS_OPTS_SCALABLE_B);
                break;
```

If we delete that, it will come on line 846, check it is `isRMW` and go inside `if` block and call `emitIns_R_R()`. Unless I am missing something major here.

Reply:
From the line you are talking about, when the `opt` gets passed to `emitIns_R_R`, the `opt` is not guaranteed to be `INS_OPTS_SCALABLE_B`

Reply:
Ok, that's where the confusion was on my part. I was looking at the unit tests and thought they both take just `INS_OPTS_SCALABLE_B`, but looking again at https://docsmirror.github.io/A64/2023-06/pnext_p_p_p.html, it can get other values as well. Makes sense now. Thanks!
 
https://github.com/dotnet/runtime/blob/7f7bb7b1530b85ec8d0d4b6e5f5429a6b729b076/src/coreclr/jit/codegenarm64test.cpp#L5047-L5048

https://github.com/dotnet/runtime/blob/7f7bb7b1530b85ec8d0d4b6e5f5429a6b729b076/src/coreclr/jit/codegenarm64test.cpp#L5034-L5035

Reply:
>  I was looking at the unit tests and thought they both take just INS_OPTS_SCALABLE_B

I did the same thing. :)